design/reboot-interface.md

```diff
@@ -0,0 +1,155 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# reboot-interface
+
+## Status
+
+provisional
+
+## Table of Contents
+
+<!--ts-->
+   * [reboot-interface](#reboot-interface)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+      * [Drawbacks](#drawbacks)
+      * [Alternatives](#alternatives)
+
+<!--te-->
+
+## Summary
+
+A declarative API is proposed to request the baremetal operator to reboot a
+provisioned Host in an atomic step (i.e. without the need to make multiple
+sequential changes to the spec).
+
+## Motivation
+
+We require a non-destructive way to fence a Kubernetes node. Some nodes cannot
+be replaced (it is currently not possible for a new master to join a cluster),
+or are expensive to replace (e.g. if this would require rebalancing Ceph data).
+A solution to this is for fencing to reboot the Host, thus ensuring that
+running processes are stopped to avoid a split-brain scenario while still
+allowing the node to rejoin the cluster with its data intact (albeit stale)
+after the reboot.
+
+The expected implementation of this on the fencing side is a Custom Resource
+Definition that requests a Machine be rebooted. The actual reboot will be
+effected either by the Machine Actuator itself (if the Cluster-API SIG can be
+persuaded that this should be part of the API in the long term), or some
+equivalent of the Machine Actuator limited to just this purpose.
+
+### Goals
+
+The Machine (Reboot) Actuator will require from the BareMetalHost:
+
+* A declarative API to perform the reboot
+* The ability to determine a point where all processes running on the Host at
+  the time of the fencing decision are guaranteed to have been stopped
+* Rapid convergence to the fenced state independent of other happenings in the
```

Comment:
Not obvious to me what "fenced state" means here - e.g. is "fenced state" the "point where  all process running on the Host are guaranteed to have been stopped"?

Reply:
If the term fencing was avoided everywhere except the "Motivation" section, it would help with clarity - I find the term often brings confusion, and replacing it with a more specific term doesn't cause any meaning to be lost

Reply:
Done